apps/web/src/pages/studio-onboarding/preview.tsx

```diff
@@ -0,0 +1,198 @@
+/* eslint-disable max-len */
+import { Prism } from '@mantine/prism';
+import { Tabs } from '@novu/design-system';
+import { IconCode, IconVisibility } from '@novu/novui/icons';
+import { css } from '@novu/novui/css';
+import { useEffect, useMemo, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { PreviewWeb } from '../../components/workflow/preview/email/PreviewWeb';
+import { useActiveIntegrations, useAuth } from '../../hooks/index';
+import { useTemplates } from '../../hooks/useTemplates';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { testTrigger } from '../../api/notification-templates';
+import { useMutation } from '@tanstack/react-query';
+import * as Sentry from '@sentry/react';
+import { IStepVariant } from '@novu/shared';
+import { api } from '../../api/index';
+import { ROUTES } from '../../constants/routes';
+import { Flex, VStack } from '@novu/novui/jsx';
+
+export const StudioOnboardingPreview = () => {
+  const { currentUser } = useAuth();
+  const [tab, setTab] = useState<string>('Preview');
+  const [content, setContent] = useState<string>('');
+  const [subject, setSubject] = useState<string>('');
+
+  const navigate = useNavigate();
+  const { integrations = [] } = useActiveIntegrations();
+  const integration = useMemo(() => {
+    return integrations.find((item) => item.channel === 'email' && item.primary) || null;
+  }, [integrations]);
+  const { mutateAsync: triggerTestEvent, isLoading } = useMutation(testTrigger);
+  const { templates, loading } = useTemplates({ pageSize: 1, areSearchParamsEnabled: false });
+
+  const template = useMemo(() => {
+    if (!templates) {
+      return null;
+    }
+
+    return templates[0];
+  }, [templates]);
+
+  const identifier = useMemo(() => {
+    if (template === null) {
+      return null;
+    }
+
+    return template.triggers[0].identifier;
+  }, [template]);
+
+  const stepId = useMemo(() => {
+    if (template === null) {
+      return null;
+    }
+
+    return (template.steps[0] as IStepVariant).uuid;
+  }, [template]);
+
+  const { mutateAsync, isLoading: previewLoading } = useMutation(
+    (data) => api.post('/v1/echo/preview/' + identifier + '/' + stepId, data),
+    {
+      onSuccess(data: any) {
+        setContent(data.outputs.body);
+        setSubject(data.outputs.subject);
+      },
+    }
+  );
+
+  useEffect(() => {
+    if (!identifier || !stepId) {
+      return;
+    }
+
+    mutateAsync();
+  }, [identifier, mutateAsync, stepId]);
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** This seems like it could all be combined as an async function outside of React (no memos necessary), which would help readability and testability. Could we consider going that route?‏